# [\#4666 Issue](https://github.com/ethereum/solidity/issues/4666) `closed`: Can set but not get when calling another contract on private chain
**Labels**: `waiting for more input`


#### <img src="https://avatars.githubusercontent.com/u/17177377?u=45b903f9af34ac0783a81c8148900696ba74bb8f&v=4" width="50">[lirycoac](https://github.com/lirycoac) opened issue at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666):

Here's my code, including the caller and called contract:

	pragma solidity ^0.4.24;

	contract Called {
	    string info = "default";

	    function setInfo(string _info) public {
	        info = _info;
	    }

	    function getInfo() view public returns (string) {
	        return info;
	    }
	}

	contract Caller {
	    Called called;

	    constructor(address addr) public {
	        called = Called(addr);
	    }

	    function setInfo(string _info) public {
	        called.setInfo(_info);
	    }

	    function getInfo() view public returns (string) {
	        return called.getInfo();
	    }
	}

There are two scenarios:

**First**, I deployed the contracts on my private chain and I set the gas price to be zero on all transactions. Both the setter and getter works well when I invoke them **directly in the Called** contract. But when I do it through the **Caller** contract, the setter works fine while the getter doesn't.

- When I invoked getter **with** keyword "view", no error occured but the result I got was a empty string(when the variable is int, the result was 0).
- when I invoked getter **without** keyword "view", the error message was: Internal JSON-RPC error.

**For the second scenario**, I deployed them on to Ropsten Test Net and set gas price to be 1 gwei on each transaction. **Everything just worked fine**.

BTW, I'm using Remix online IDE and MetaMask chrome extension.

#### <img src="https://avatars.githubusercontent.com/u/30042495?v=4" width="50">[zhaohongyin](https://github.com/zhaohongyin) commented at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666#issuecomment-410197593):

I deployed your contracts in Remix online IDE.. 
Called contract works well but the Caller contract just can get and can not set
here is the error message Error: VM Exception while processing transaction: revert 
BTW, I used ganache and metamask

#### <img src="https://avatars.githubusercontent.com/u/9073706?v=4" width="50">[chriseth](https://github.com/chriseth) commented at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666#issuecomment-410203875):

Did you set the EVM version of your executing environment to `Byzantium`? The recent versions of the compiler use some features that are not present in the older EVM versions.

#### <img src="https://avatars.githubusercontent.com/u/17177377?u=45b903f9af34ac0783a81c8148900696ba74bb8f&v=4" width="50">[lirycoac](https://github.com/lirycoac) commented at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666#issuecomment-410564795):

@chriseth , Sorry for the delay. I've tried setting the compiler version in Solidity to 0.4.18 and it worked. Thanks!

#### <img src="https://avatars.githubusercontent.com/u/10923091?u=8297295cdc0493d4bd9b5ba4711c3b07c5c7b836&v=4" width="50">[tanhuiya](https://github.com/tanhuiya) commented at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666#issuecomment-443083060):

what about the version after 0.4.18 ?

#### <img src="https://avatars.githubusercontent.com/u/9073706?v=4" width="50">[chriseth](https://github.com/chriseth) commented at [2018-08-03 03:23](https://github.com/ethereum/solidity/issues/4666#issuecomment-443126495):

@tanhuiya please open a new issue with detailed description.


-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2022.05.23 at 03:51:38.]
